### ISSUE TITLE
BTCPay 2.0.0-alpha integrations testing

### ISSUE THREAD

**Issue by ndeet** [btcpayserver/btcpayserver#6200]
### What is your BTCPay version?

v2.0.0-alpha2

### How did you deploy BTCPay Server?

docker

### What happened?

Testing WooCommerce and other integrations and collect breaking changes here:

1. LN payment does not pay the full amount but it says "has not been paid in full" when there is a small deviation on 12th decimal point

![image](https://github.com/user-attachments/assets/d90daab4-cd4d-4b11-b31d-3b19ba6e0c04)

### How did you encounter this bug?

pay LN invoice

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by ndeet** [comment]
@NicolasDorier the `config` data is missing on the Store PaymentMethods endpoint, it just gives me this:
```
[
    {
        "enabled": true,
        "paymentMethodId": "BTC-LN"
    },
    {
        "enabled": true,
        "paymentMethodId": "BTC-CHAIN"
    },
    {
        "enabled": true,
        "paymentMethodId": "BTC-LNURL"
    }
]
```
on the old datastructure `config` was called `data`, do you think we can keep `data`  or should we stick with `config`?

old data structure:
```
{
    "BTC": {
        "enabled": true,
        "data": {
            "derivationScheme": "tpubD6NzVbkrYhZ4YjkqGkhJB8BCYpy31J5YdSWyJ9p5v9mVadicqyjLgCnL4CyFP3NbQeAR1gci8MhYGQM7dWyr9C17UFBhDrkdJ2BRFzhx5bf-[legacy]",
            "label": "btc testnet",
            "accountKeyPath": null
        },
        "cryptoCode": "BTC"
    },
    "BTC-LNURLPAY": {
        "enabled": true,
        "data": {
            "useBech32Scheme": false,
            "lud12Enabled": false
        },
        "cryptoCode": "BTC"
    },
    "BTC-LightningNetwork": {
        "enabled": true,
        "data": {
            "connectionString": "Internal Node"
        },
        "cryptoCode": "BTC"
    }
}
```

**Comment by dennisreimann** [comment]
@ndeet See the `includeConfig` query parameter for that one.

**Comment by ndeet** [comment]
Oh I missed that one, thank you @dennisreimann 

Need to check how I can make it work without a breaking change in PHP library. I see why it now has `canmodifystoresettings` permission but need to figure out how to make it work with the `canviewstoresettings` permission without breaking anything in integrations.

**Comment by ndeet** [comment]
@NicolasDorier what you think about changing `config` to `data` to be consistent with < 2.0?

**Comment by ndeet** [comment]
@NicolasDorier any reason why the `currency` is missing on [Store Payment Methods](https://testing.btcpay.tech/docs#operation/StorePaymentMethods_GetStorePaymentMethods), it is there on [Invoice Payment Methods](https://testing.btcpay.tech/docs#operation/Invoices_GetInvoicePaymentMethods) and for consitency we should make it available also on the main payment methods endpoint?

**Comment by NicolasDorier** [comment]
@ndeet there isn't any currency in store payment's method, because the currency is decided when the payment method create a prompt for the invoice. A payment method can have several currency possible. (though we don't use it now)

**Comment by dennisreimann** [comment]
@ndeet Thanks for the pointer, fixed in e389d6a96b3b657e5ca02508a97f138f41f5163c.

**Comment by ndeet** [comment]
@NicolasDorier on the [Invoices -> Get Payment Methods endpoint](https://testing.btcpay.tech/docs#operation/Invoices_GetInvoicePaymentMethods) in `additionalData` are you aware we leak the xpub? It only requires the `canviewinvoices` permission.

I ask because on the [Store Payment Methods endpoint](https://testing.btcpay.tech/docs#operation/StorePaymentMethods_GetStorePaymentMethods) the config is protected by `canmodifystoresettings` permission when you use the `includeConfig` parameter.

So if it is protected there then we should not leak it here with less permissions?

```
...
["paymentMethodId"]=>
      string(9) "BTC-CHAIN"
      ["additionalData"]=>
      array(5) {
        ["keyPath"]=>
        string(3) "0/1"
        ["payjoinEnabled"]=>
        bool(true)
        ["accountDerivation"]=>
        string(111) "xpub6DVMcQAQCtGbNDTEjQGtR1GRoTKw7AzP6bVivX4gFnewcnRk1r1tbczpfsaYjKKVrmtyiwYqAEnALYzZ8yoTArVsKfZekmwLFqQp4MRgPhy"
        ["recommendedFeeRate"]=>
        float(4.107)
        ["paymentMethodFeeRate"]=>
        float(3.975)
      }
```

**Comment by ndeet** [comment]
Collecting updated integrations:

- Released a BTCPay 2.0 compatible version of the PHP library which also works with current and older versions 
- Released new BTCPay 2.0 compatible WooCommerce integration https://github.com/btcpayserver/woocommerce-greenfield-plugin/releases/tag/v2.7.0

**Comment by NAKKIABW** [comment]
Where can i find the download for the wallet for btcpay
